pkg/utils/metrics.go

```diff
@@ -45,21 +46,126 @@ func NewAgentMetrics(ai *AgentInfo) AgentMetrics {
 		Help:        "Total number of reports (keepalive messages) from the agent.",
 	})
 
-	if counter, ok := tryRegister(am.ErrorCount); !ok {
+	if counter, ok := tryRegisterCounter(am.ErrorCount); !ok {
 		// use existing counter
 		am.ErrorCount = counter
 	}
-	if counter, ok := tryRegister(am.ReportCount); !ok {
+	if counter, ok := tryRegisterCounter(am.ReportCount); !ok {
 		// use existing counter
 		am.ReportCount = counter
 	}
 
+  am.ProbeConnectionResult = prometheus.NewGaugeVec(
```

Comment:
Maybe use reflection and do NewGaugeVec within a loop?
E.g., https://github.com/Mirantis/k8s-netchecker-agent/blob/master/agent.go#L214